src/systems/button-systems.js

```diff
@@ -0,0 +1,83 @@
+export class SingleActionButtonSystem {
+  tick() {
+    const interaction = AFRAME.scenes[0].systems.interaction;
+    const userinput = AFRAME.scenes[0].systems.userinput;
+    const state = interaction.state.rightRemote;
+    const grab = interaction.options.rightRemote.grabPath;
+    if (
+      userinput.get(grab) &&
+      state.hovered &&
+      state.hovered.components.tags &&
+      state.hovered.components.tags.data.singleActionButton
+    ) {
+      state.hovered.object3D.dispatchEvent({
+        type: "interact",
+        path: grab
+      });
+    }
+  }
+}
+
+export class HoldableButtonSystem {
+  tick() {
+    const interaction = AFRAME.scenes[0].systems.interaction;
+    const held = interaction.state.rightRemote.held;
+    const options = interaction.options.rightRemote;
+
+    if (this.prevHeld && this.prevHeld !== held) {
+      this.prevHeld.object3D.dispatchEvent({
+        type: "holdable-button-up",
+        path: options.dropPath
+      });
+    }
+    if (held && this.prevHeld !== held) {
+      held.object3D.dispatchEvent({
+        type: "holdable-button-down",
+        path: options.grabPath
+      });
+    }
+
+    this.prevHeld = held;
+  }
+}
+
+function getHoverableButton(hovered) {
+  if (!hovered) return null;
+  if (
+    hovered.components["icon-button"] ||
+    hovered.components["text-button"] ||
+    hovered.components["pin-networked-object-button"]
+  )
+    return hovered;
+  // TODO: fix this so that we aren't looping thru children here. I just did this to accomodate the new rounded buttons
```

Comment:
The rounded buttons I'm referring to here are the ones that got added here: https://github.com/mozilla/hubs/pull/1124/files#diff-a796e11be3d0dfc0f893148f1086736dR245

There is probably a smarter way I could be doing this but I don't know what it is.